### ISSUE TITLE
Misleading comment about  UniqueJsonKeyClaimAction

### ISSUE THREAD

**Issue by LeaFrock** [dotnet/aspnetcore#38572]
## Summary

It's just a discussion/suggestion about the comment of `UniqueJsonKeyClaimAction`. 

## Motivation and goals

The comment on second line of `UniqueJsonKeyClaimAction` says,
>/// This no-ops if the ClaimsIdentity already contains a Claim with the given ClaimType.

But the source code is that,
```csharp
        var claim = identity.FindFirst(c => string.Equals(c.Type, JsonKey, System.StringComparison.OrdinalIgnoreCase));
        if (claim != null && string.Equals(claim.Value, value, System.StringComparison.Ordinal))
        {
            // Duplicate
            return;
        }
```
It no-ops if the ClaimsIdentity already contains a Claim with the given **JsonKey** instead of ClaimType.

If there're some claims like `name`(which ` OpenIdConnectOptions`'s constructor initializes) , in the following codes,
```csharp
//Scenary 1: Should no-op?
option.ClaimActions.MapUniqueJsonKey(claimType:"name", jsonKey:"custom_json_key"); 

//Scenary 2: Want another claim which is from JsonKey `name` and has the same value as ClaimType `name`. But it no-ops?
option.ClaimActions.MapUniqueJsonKey(claimType:"custom_claim_type", jsonKey:"name"); 
```
I think it's a little bit confusing. Or is it a misunderstanding of me?

## Risks / unknowns

None.

**Comment by Tratcher** [comment]
The comment is correct, but the code is wrong 😁.

In the common case the ClaimType and the JsonKey are the same:
https://github.com/dotnet/aspnetcore/blob/8d559c6c1a1d006807978310d2e685f53082dd8f/src/Security/Authentication/OpenIdConnect/src/OpenIdConnectOptions.cs#L63-L68

But you're right that they don't have to be the same, in which case ClaimType should be used for these lookups, not JsonKey.
https://github.com/dotnet/aspnetcore/blob/8d559c6c1a1d006807978310d2e685f53082dd8f/src/Security/Authentication/OpenIdConnect/src/UniqueJsonKeyClaimAction.cs#L39-L51

Since that's what's used to actually create the claim:
https://github.com/dotnet/aspnetcore/blob/8d559c6c1a1d006807978310d2e685f53082dd8f/src/Security/Authentication/OpenIdConnect/src/UniqueJsonKeyClaimAction.cs#L58

I'll see about fixing it.

**Comment by LeaFrock** [comment]
@Tratcher Well... so accidentally I found a bug 😆.
Will it be fixed in ASP.NET Core 6?

**Comment by Tratcher** [comment]
The fix has been checked into 7.0. It won't be backported to 6.0 unless someone can demonstrate it has a significant impact and is hard to work around. In this case the workarounds are easy enough. Worst case you can copy the class.
